plugins/TPReplayModule.cpp

```diff
@@ -0,0 +1,551 @@
+    m_filter_planes_ids.push_back(plane_wrap->get_plane());
+  }
+  m_filter_planes = (m_filter_planes_ids.size() > 0) ? true : false;
+
+  TLOG() << "Plane filtering: " << m_filter_planes;
+  if (m_filter_planes) {
+    TLOG() << "Planes to filter: ";
+    for (auto plane : m_filter_planes_ids) {
+      TLOG() << plane;
+    }
+  }
+
+  // For each of the files that are specified in the config, we extract data and sort them.
+  // Data is sorted by ROU -> Plane (if not filtered). Multiple files can contribute to each.
+  // Then we create an outgoing sink for each unique ROU + plane combination.
+  // We also keep track of the total timestamp range of all the streams, so we can keep
+  // the timestamps of the multiple streams in sync when replaying,
+  // even when they don't all start or end at the same time.
+
+  // Output queues
+  auto con = mtrg->get_outputs();
+
+  // Global times
+  m_earliest_first_tp_timestamp = std::numeric_limits<triggeralgs::timestamp_t>::max();
+  m_latest_last_tp_timestamp = 0;
+
+  // Loading sorted TP stream files
+  for (auto& stream : m_conf->get_tp_streams()) {
+    std::pair tmp_obj = std::make_pair(stream->get_index(), stream->get_filename());
+    m_tpstream_files.insert(tmp_obj);
+  }
+
+  // Print grouped files
+  TLOG() << "Files to use:";
+  for (const auto& pair : m_tpstream_files) {
+    std::cout << "Index: " << pair.first << ", Filename: " << pair.second << std::endl;
+  }
+
+  // Load data here
+  m_all_tp_data = read_tps(m_tpstream_files);
+  if (m_tpstream_files.size() == 0 || m_all_tp_data.size() == 0) {
+    ers::error(dunedaq::trigger::ReplayNoValidFiles(ERS_HERE, get_name()));
+  }
+
+  // Data loaded and sorted.
+  // Now we create streams.
+  int global_iter = 0;
+  // Loop over ROUs
+  for (const auto& rou_pair : m_all_tp_data) {
+    const std::string& ROU = rou_pair.first;
+    const auto& plane_map = rou_pair.second;
+    int plane_iter = 0;
+    // Loop over Planes
+    for (const auto& plane_pair : plane_map) {
+      int plane = plane_pair.first;
+      const auto& vector_of_tps = plane_pair.second;
+
+      TPStream this_stream;
+      TLOG_DEBUG(1) << "Stream: " << (global_iter + plane_iter) << "; ROU: " << ROU << "; plane: " << plane
+                    << "; TP sink is " << con[global_iter + plane_iter]->class_name() << "@"
+                    << con[global_iter + plane_iter]->UID();
+      this_stream.tp_sink =
+        get_iom_sender<std::vector<trigger::TriggerPrimitiveTypeAdapter>>(con[global_iter + plane_iter]->UID());
+
+      this_stream.tpvs = vector_of_tps;
+
+      m_earliest_first_tp_timestamp =
+        std::min(m_earliest_first_tp_timestamp, this_stream.tpvs.front().front().tp.time_start);
+
+      m_latest_last_tp_timestamp = std::max(m_latest_last_tp_timestamp, this_stream.tpvs.back().back().tp.time_start);
+
+      m_tp_streams.push_back(std::move(this_stream));
+      plane_iter++;
+    }
+    global_iter = global_iter + plane_iter;
```

Comment:
Why not the standard
```suggestion
    global_iter += plane_iter;
```